clients/cli/src/prover_runtime.rs

```diff
@@ -0,0 +1,446 @@
+//! Prover Runtime
+//!
+//! Handles background execution of proof tasks in both authenticated and anonymous modes.
+//! Spawns async workers, dispatches tasks, and reports progress back to the UI.
+//!
+//! Includes:
+//! - Task fetching from the orchestrator (authenticated mode)
+//! - Worker management and task dispatching
+//! - Prover event reporting
+
+use crate::orchestrator::error::OrchestratorError;
+use crate::orchestrator::{Orchestrator, OrchestratorClient};
+use crate::prover::authenticated_proving;
+use crate::task::Task;
+use chrono::Local;
+use ed25519_dalek::{SigningKey, VerifyingKey};
+use nexus_sdk::stwo::seq::Proof;
+use sha3::{Digest, Keccak256};
+use std::time::Duration;
+use tokio::sync::{broadcast, mpsc};
+use tokio::task::JoinHandle;
+
+/// Events emitted by prover (worker) threads.
+#[allow(unused)]
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum WorkerEvent {
+    TaskFetcher { data: String },
+    Prover { worker_id: usize, data: String },
+    ProofSubmitter { data: String },
+}
+
+/// Starts authenticated workers that fetch tasks from the orchestrator and process them.
+pub async fn start_authenticated_workers(
+    node_id: u64,
+    signing_key: SigningKey,
+    orchestrator: OrchestratorClient,
+    num_workers: usize,
+    shutdown: broadcast::Receiver<()>,
+) -> (mpsc::Receiver<WorkerEvent>, Vec<JoinHandle<()>>) {
+    let mut join_handles = Vec::new();
+    // Worker events
+    let (event_sender, event_receiver) = mpsc::channel::<WorkerEvent>(100);
```

Comment:
I assume this parameter follows from the `task_queue_size` below? If so shouldn't it use that constant? If it's a coincidence, that should be documented (as with all magic numbers).